app.go

```diff
@@ -767,7 +767,15 @@ type withTimeoutParams struct {
 
 func withTimeout(ctx context.Context, param *withTimeoutParams) error {
 	c := make(chan error, 1)
-	go func() { c <- param.callback(ctx) }()
+	go func() {
+		// If runtime.Goexit() is called from within the callback
+		// then nothing is written to the chan.
+		// However the defer will still be called, so we can write to the chan,
+		// to avoid hanging until the timeout is reached.
+		// If the callback returned normally then our deferred write will never be read.
+		defer func() { c <- fmt.Errorf("withTimeout exit without returning") }()
+		c <- param.callback(ctx)
+	}()
```

Comment:
changed to satisfy both your suggestions, I don't think its strictly necessary as there isn't a race in the normal scenario, but with the boolean it might prevent odd future issues and it's more clear what the intention of the code is as well.